flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -94,29 +108,91 @@ public class AsyncExecutionController<K> {
      */
     final AtomicInteger inFlightRecordNum;
 
+    /**
+     * The flag to indicate whether the {@link #bufferTimeOut} is reached, if yes, a trigger will
+     * perform actively when the next state request arrives even if the activeQueue has not reached
+     * the {@link #batchSize}.
+     */
+    final AtomicBoolean timeoutFlag;
+
+    /** The executor service that schedules and calls the triggers of this task. */
+    final ScheduledThreadPoolExecutor scheduledExecutor;
+
+    ScheduledFuture<Void> currentScheduledFuture;
+
     public AsyncExecutionController(MailboxExecutor mailboxExecutor, StateExecutor stateExecutor) {
-        this(mailboxExecutor, stateExecutor, DEFAULT_BATCH_SIZE, DEFAULT_MAX_IN_FLIGHT_RECORD_NUM);
+        this(
+                mailboxExecutor,
+                stateExecutor,
+                DEFAULT_BATCH_SIZE,
+                DEFAULT_BUFFER_TIMEOUT,
+                DEFAULT_MAX_IN_FLIGHT_RECORD_NUM);
     }
 
     public AsyncExecutionController(
             MailboxExecutor mailboxExecutor,
             StateExecutor stateExecutor,
             int batchSize,
+            long bufferTimeOut,
             int maxInFlightRecords) {
         this.keyAccountingUnit = new KeyAccountingUnit<>(maxInFlightRecords);
         this.mailboxExecutor = mailboxExecutor;
         this.stateFutureFactory = new StateFutureFactory<>(this, mailboxExecutor);
         this.stateExecutor = stateExecutor;
         this.batchSize = batchSize;
+        this.bufferTimeOut = bufferTimeOut;
```

Comment:
The "O` in "BufferTimeOut` is upper-case while the "o" in timeoutFlag is lower-case. It might be better to get them unified to the same convention.